src/Neo/SmartContract/Native/LedgerContract.cs

```diff
@@ -107,7 +113,10 @@ public UInt256 GetBlockHash(DataCache snapshot, uint index)
         [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
         public UInt256 CurrentHash(DataCache snapshot)
         {
-            return snapshot[CreateStorageKey(Prefix_CurrentBlock)].GetInteroperable<HashIndexState>().Hash;
+            if (snapshot is null)
+                throw new ArgumentNullException(nameof(snapshot));
+
+            return snapshot[CreateStorageKey(Prefix_CurrentBlock)]?.GetInteroperable<HashIndexState>()?.Hash;
```

Comment:
Don't change the behavior null is not the previous logic

Reply:
i agree, this should not be changed to nullable.